src/core/SIP/SIPMessageBuffer.cs

```diff
@@ -36,11 +36,25 @@ public class SIPMessageBuffer
 
         private static ILogger logger = Log.Logger;
 
-        public string RawMessage;
+        public string RawMessage
+        {
+            get
+            {
+                if (RawBuffer != null)
+                {
+                    return Encoding.UTF8.GetString(RawBuffer);
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
         public SIPMessageTypesEnum SIPMessageType = SIPMessageTypesEnum.Unknown;
         public string FirstLine;
         public string[] SIPHeaders;
-        public string Body;
+        public byte[] Body;
         public byte[] RawBuffer;
```

Comment:
The body is stored twice. Both in this byte[] body and also exists in RawBuffer. Maybe add indexes to where the body bytes are within the RawBuffer to have less memory allocation?

Reply:
But even if the `Body` buffer was changed to a property a new `byte[]` array would still need to be allocated whenever it was accessed. Or can you see another approach?

The ideal way to deal with this type of situation is a `ReadOnlySpan<byte>` to provide a slice of the `RawBuffer` array without needing any additional allocations. The problem is  `ReadOnlySpan's` requires dropping .NET Framework support (or additional dependencies) and it's not worth the trade-off at this point.

Reply:
SipRequest and SipResponse could have a reference to a SIPMessageBuffer to store it's SIPMessage and no copying of data would be nessesary, but that is a subsequent PR. The Body reference allocated is passed on to SipRequest or SipResponse so have it allocated here or later when copied to the SipRequest or SipResponse doesn't make that much a difference. Leave it like this, I think.